### PR TITLE
[RHOAIENG-6867] Tracking of dashboard version and users's capabilities

### PR DESCRIPTION
**Pull request by pilhuhn**
## Description

This is related to [RHOAIENG-6867](https://issues.redhat.com/browse/RHOAIENG-6867)

* Add dashboard version to track events / page views
* On user init record the capabilities of the user 

## How Has This Been Tested?
Manually be looking at the data that arrived in Segment / by looking at request in the Browsers network tab

## Test Impact
No change on functionality of the visible UI

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/utilities/segmentIOUtils.tsx [hunk 1]
```diff
@@ -126,7 +193,7 @@ export const initSegment = async (props: {
     );
     const anonymousIDArray = Array.from(new Uint8Array(anonymousIDBuffer));
     const anonymousID = anonymousIDArray.map((b) => b.toString(16).padStart(2, '0')).join('');
-    fireTrackingEvent('identify', { anonymousID });
+    fireIdentifyEvent({ anonymousID, canCreateProjects: canCreate, canDelete, canUpdate, isAdmin });
```
Comment (andrewballantyne) [inline frontend/src/utilities/segmentIOUtils.tsx:196]: Why did you rename "canCreate" and not the other props?

File: frontend/src/utilities/segmentIOUtils.tsx [hunk 2]
```diff
@@ -15,7 +15,49 @@ export const fireTrackingEventRaw = (eventType: string, properties?: any): void
...
+export type IdentifyProps = {
+  anonymousID: string;
+  canCreateProjects: boolean;
+  canDelete: boolean;
+  canUpdate: boolean;
+  isAdmin: boolean;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any,@typescript-eslint/explicit-module-boundary-types
+export const fireIdentifyEvent = (properties?: IdentifyProps): void => {
```
Comment (andrewballantyne) [inline frontend/src/utilities/segmentIOUtils.tsx:42]: What explicit any are you preventing?

Comment (pilhuhn) [reply frontend/src/utilities/segmentIOUtils.tsx:42]: Leftover from when I started the method. Good catch.

File: frontend/src/utilities/segmentIOUtils.tsx (hunk 1 above)
Comment (pilhuhn) [reply frontend/src/utilities/segmentIOUtils.tsx]: Also a leftover from when this started. I've changed this now.

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
I'm noticing some extra stuff here as I'm trying to look at approving your work here -- see my new review questions.

Also -- We'll need to look to get you into our scrum. Your ticket is now moved to the sprint but you're going to need to attend and communicate through our scrum.

Please direct any questions to our UXD Coordinator, @jeff-phillips-18 
cc @dgutride

File: frontend/src/utilities/useSegmentTracking.ts [hunk 3]
```diff
@@ -10,14 +13,66 @@ export const useSegmentTracking = (): void => {
...
+      clusterID
+    ) {
       window.clusterID = clusterID;
       initSegment({
         segmentKey,
         username,
         enabled: !dashboardConfig.spec.dashboardConfig.disableTracking,
+        canCreate: allowCreate,
+        canDelete: allowDelete,
+        canUpdate: allowUpdate,
```
[the remark this answers is not in the document]
Comment (pilhuhn) [reply frontend/src/utilities/useSegmentTracking.ts:59]: The analytics.identify() happens within init. And identify() is the place to send user-properties. 
We can for sure send that any time later. As this is only needed once and not on every page load, init looks like a good place for me.

**Comment by pilhuhn** [comment]
I've broken out the dashboard version part in https://github.com/opendatahub-io/odh-dashboard/pull/2896

File: frontend/src/utilities/useSegmentTracking.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (andrewballantyne) [reply frontend/src/utilities/useSegmentTracking.ts:59]: Our segment tracking was haphazardly put together and is still in rough shape. The more you add to it the more I want to spend time making sure we are doing the right thing, structured and future looking.

`init` is init of the infrastructure and should be a cohesive set of "startup and mounting of segment" statements. It shouldn't manage the initial state of the logged in user, that should be done via another hook once things have started up.

I'd like to start seeing proper documentation and setup for these things... maybe even a `src/concepts/segment` folder where we build out utilities to use it throughout our app.

**Comment by pilhuhn** [comment]
I am closing this in favor of a better solution going forward.
